pkgs/applications/networking/irc/weechat/default.nix

```diff
@@ -117,13 +118,34 @@ in if configure == null then weechat else
         ln -s $plugin $out/plugins
       done
     '';
-  in (writeScriptBin "weechat" ''
-    #!${stdenv.shell}
-    export WEECHAT_EXTRA_LIBDIR=${pluginsDir}
-    ${lib.concatMapStringsSep "\n" (p: lib.optionalString (p ? extraEnv) p.extraEnv) plugins}
-    exec ${weechat}/bin/weechat "$@"
-  '') // {
-    name = weechat.name;
-    unwrapped = weechat;
-    meta = weechat.meta;
+
+    init = let
+      init = builtins.replaceStrings [ "\n" ] [ ";" ] (config.init or "");
+      scripts = builtins.concatStringsSep ";" (map (s: "/script load ${s}") (config.scripts or []));
+    in init + scripts;
+
+    mkWeechat = bin: (writeScriptBin bin ''
+      #!${stdenv.shell}
+      index=0
+      next=
+      declare -a list=("$@")
+      for i in "$@"; do
+        ((index+=1))
+        if [[ "$i" = "--run-command" || "$i" = "-r" ]]; then
+          next="$next;''${list[$index]}"
```

Comment:
We probably shouldn't modify weechats standard behaviour of overriding previous `--run-commands` (that were explicitly passed by the user). Although I personally think aggregating them would make way more sense.

On the other hand weechat may change that in the future and there isn't really too much harm in aggregating them (who in their right mind would depend on the override behaviour?), so I'm not sure.

Reply:
I've opened https://github.com/weechat/weechat/issues/1248 upstream for that.

Reply:
You currently append to `next` each time you find a `--run-command`, which is different from the default weechat behaviour.

However somebody upstream already said they'll "implement that quickly", so it should be fine as-is.